kbc/parser.js

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env node
+
+'use strict';
+const Parser = require('../.lib/parser.js');
+
+/**
+ * Recognizes the accesses to the platform Know BC
+ * @param  {Object} parsedUrl an object representing the URL to analyze
+ *                            main attributes: pathname, query, hostname
+ * @param  {Object} ec        an object representing the EC whose URL is being analyzed
+ * @return {Object} the result
+ */
+module.exports = new Parser(function analyseEC(parsedUrl, ec) {
+  let result = {};
+  let path   = parsedUrl.pathname;
+  // uncomment this line if you need parameters
+  // let param = parsedUrl.query || {};
+
+  // use console.error for debuging
+  // console.error(parsedUrl);
+
+  let match;
+
+  if ((match = /^\/books\/([a-zA-Z-]+)\/([a-zA-Z-]+)\/([a-zA-Z-]+)\/([a-zA-Z-]+)(:?\/([a-zA-Z0-9-]+))?$/i.exec(path)) !== null) {
```

Comment:
A non-capturing group is marked by a `?:`, not by a `:?`
```suggestion
  if ((match = /^\/books\/([a-zA-Z-]+)\/([a-zA-Z-]+)\/([a-zA-Z-]+)\/([a-zA-Z-]+)(?:\/([a-zA-Z0-9-]+))?$/i.exec(path)) !== null) {
```